frontends/web/src/components/dialog/dialog.tsx

```diff
@@ -156,12 +156,25 @@ class Dialog extends Component<Props, State> {
       clearTimeout(this.timerId);
     }
 
-    this.modal.current?.classList.remove(style.open);
     this.overlay.current.classList.remove(style.activeOverlay);
     this.overlay.current.classList.add(style.closingOverlay);
-    this.timerId = setTimeout(() => {
+    this.modal.current?.classList.remove(style.open);
+
+    const onTransitionEnd = (event: TransitionEvent) => {
+      if (event.target === this.modal.current) {
+        this.deactivateModal(fireOnCloseProp);
+        this.modal.current?.removeEventListener('transitionend', onTransitionEnd);
+      }
```

Comment:
on small screen I behaves correctly but I get this JS error:

![Screen Shot 2023-11-30 at 20 56 33](https://github.com/digitalbitbox/bitbox-wallet-app/assets/546900/60499117-6b3b-4c6e-930a-81c41954e5f4)